src/services/clipService.ts

```diff
@@ -220,3 +323,119 @@ export async function computeTextEmbedding(
         throw err;
     }
 }
+
+export async function computeONNXTextEmbedding(
+    text: string
+): Promise<Float32Array> {
+    try {
+        const imageSession = await getOnnxTextSession();
+        const tokenizer = getTokenizer();
+        const tokenizedText = Int32Array.from(tokenizer.encodeForCLIP(text));
+        const feeds = {
+            input: new ort.Tensor('int32', tokenizedText, [1, 77]),
+        };
+        const results = await imageSession.run(feeds);
+        const embedVec = results['output'].data; // Float32Array
+        return normalizeEmbedding(embedVec);
+    } catch (err) {
+        if (err.message === CustomErrors.MODEL_DOWNLOAD_PENDING) {
+            log.info(CustomErrors.MODEL_DOWNLOAD_PENDING);
+        } else {
+            logErrorSentry(err, 'Error in computeTextEmbedding');
+        }
+        throw err;
+    }
+}
+
+async function getRGBData(inputFilePath: string) {
+    const jpegData = await readFile(inputFilePath);
+    const rawImageData = jpeg.decode(jpegData, {
```

Comment:
Also maybe when the jpeg decode fails too?

Reply:
no point in trying to catch the error, if we can't do anything for it?

Reply:
Maybe to understand how many of the thumbnails are PNG? I don't know just wanted it to be complete. Your call here